### ISSUE TITLE
Xcode 13 RC (13A233) support for armv7 (Real devices)

### ISSUE THREAD

**Issue by niorko** [grpc/grpc-swift#1271]
### Describe the bug

Xcode version: 13 RC (13A233)
Target: GRPC - Any iOS Device (or real iPhone - not simulator)
Project version: 1.4.1 (66e27d7e) - standalone SPM Package.swift

The project is not buildable for real iPhone devices (or Any iOS Device).
A simulator target builds successfully.

### To reproduce

Steps to reproduce the bug you've found:
1. Clone 1.4.1 (66e27d7e)
2. Choose GRPC Target with **Any iOS Device**
3. Build
4. -> Build failed

### Expected behaviour

Build succeeds.

### Additional information

Build log: [Build GRPC_2021-09-17T08-39-43.txt](https://github.com/grpc/grpc-swift/files/7183323/Build.GRPC_2021-09-17T08-39-43.txt)

**Comment by Lukasa** [comment]
This is a very interesting set of errors: they all come from SwiftNIO SSL, but they don't appear when building that project directly so far as I can see. Additionally, the nature of these errors is that there appears to be a header mismatch: we seem to be getting the wrong header files from CNIOBoringSSL.

Can I confirm that this occurs for you on a clean build (that is, if you `rm -rf .build`)?

**Comment by niorko** [comment]
It happens even for clean build. But when I select any simulator, build succeed.
And it does not work with real device or in particular _Any iOS Device_.

I tested it on both my machines (Intel MacMini 2018 and MacbookPro 2018 both on macOS 11.5.2).

**Comment by Lukasa** [comment]
It should work with real devices using arm64. The issue seems likely to be to do with the way armv7 is building.

**Comment by Lukasa** [comment]
The reason I'm surprised this has anything to do with armv7 is that the errors are primarily the result of whether the `X509` structure is opaque or not. This should not be architecture specific.

**Comment by niorko** [comment]
[Arm64 - ❌](https://github.com/grpc/grpc-swift/files/7183905/Build.GRPC_2021-09-17T10-15-18.txt)
[Simulator - ✅](https://github.com/grpc/grpc-swift/files/7183918/Build.GRPC_2021-09-17T10-16-10.txt)

I including build from arm64 and it does not build.

**Comment by Lukasa** [comment]
Ah, this second log is more useful. This seems to show a problem with the modulemaps: we're including the same header file through separate paths. I have to admit I don't know why this is causing some of these issues: the compiler seems to be having real trouble resolving a bunch of the includes.

**Comment by Lukasa** [comment]
For example:

```
/Volumes/FBMac/tmp/DerivedData/kate-ios-grpc-swift-aeudckuyjkzqicdqqcwbtlwoieon/SourcePackages/checkouts/swift-nio-ssl/Sources/NIOSSL/SSLContext.swift:401:20: error: ambiguous use of 'CNIOBoringSSL_SSL_CTX_use_PrivateKey_file'
            return CNIOBoringSSL_SSL_CTX_use_PrivateKey_file(context, pointer, fileType)
                   ^
CNIOBoringSSL.CNIOBoringSSL_SSL_CTX_use_PrivateKey_file:1:13: note: found this candidate
public func CNIOBoringSSL_SSL_CTX_use_PrivateKey_file(_ ctx: OpaquePointer!, _ file: UnsafePointer<CChar>!, _ type: Int32) -> Int32
            ^
CNIOBoringSSLShims.CNIOBoringSSL_SSL_CTX_use_PrivateKey_file:1:13: note: found this candidate
public func CNIOBoringSSL_SSL_CTX_use_PrivateKey_file(_ ctx: OpaquePointer!, _ file: UnsafePointer<CChar>!, _ type: Int32) -> Int32
```

This shows the function `CNIOBoringSSL_SSL_CTX_use_PrivateKey_file` being available from `CNIOBoringSSL` _and_ `CNIOBoringSSLShims`. That shouldn't be the case, we should be resolving only the one.

However, we also see errors like this:

```
/Volumes/FBMac/tmp/DerivedData/kate-ios-grpc-swift-aeudckuyjkzqicdqqcwbtlwoieon/SourcePackages/checkouts/swift-nio-ssl/Sources/NIOSSL/SSLContext.swift:413:90: error: ambiguous use of 'count'
            CNIOBoringSSL_SSL_CTX_set_alpn_protos(context, $0.baseAddress!, CUnsignedInt($0.count))
                                                                                         ^
Swift.UnsafeBufferPointer:2:16: note: found this candidate
    public let count: Int
               ^
Swift.Collection:5:27: note: found this candidate
    @inlinable public var count: Int { get }
```

This shouldn't really be ambiguous. For now I'm assuming this error is an awkward follow-on from the former, but it could be something else altogether.

**Comment by Lukasa** [comment]
So far I still haven't reproduced this. Can you remove your `.build` _and_ your `DerivedData` folder to just be 100% sure that we aren't bringing in some weird state from elsewhere?

**Comment by niorko** [comment]
I did it, and it still didn't work on the MacMini. I have my DervivedData and products with indexes set to be on the external SSD, due to small capacity of the MacMini, so it seems like an Xcode issue.

I did a clean copy with .build and DerviceData removed on the secondary Macbook and it build successfully.

Sorry for the issue and thank you for your time and effort. If I find anything new or I better isolate the issue on MacMini, I could let you know. In the meantime I am going to close the issue.
